internal/cli/command/controlplane/installer_options.go

```diff
@@ -239,6 +244,49 @@ func (c *cpContext) readEc2Host() (string, error) {
 	return strings.Trim(string(bytes), "\n"), nil
 }
 
+func (c *cpContext) logDir() (string, error) {
+	dir := filepath.Join(c.workspace, logsDir)
+	if err := os.MkdirAll(dir, 0777); err != nil {
+		return "", errors.WrapIf(err, "failed to create log directory")
+	}
+
+	return dir, nil
+}
+
+func (c *cpContext) createLog(nameParts ...string) (io.WriteCloser, error) {
+	if !c.logOutput {
+		return nil, nil
+	}
+
+	logdir, err := c.logDir()
+	if err != nil {
+		return nil, err
+	}
+
+	ts, _ := time.Now().MarshalText()
+	name := fmt.Sprintf("%s-%s.log", ts, strings.ReplaceAll(strings.Join(nameParts, "-"), "/", ""))
+	return os.Create(filepath.Join(logdir, name))
+}
+
+func (c *cpContext) listLogs() (dir string, out []string, err error) {
+	dir, err = c.logDir()
+	if err != nil {
+		return
```

Comment:
**Note**: I would prefer `return "", nil, err` in these cases for readability and a `return dir, out, nil` at the end of the function, because otherwise I have to deep-scan the function body to find the last place where those variables were set individually and for that I also have to check the function signature or keep the names of the predefined return values in mind.